---
layout: page
permalink: /courses/BABS4_Biochem_DataWorkshop4_March2024
---

![BABS4 banner](/assets/images/babs4_banner.jpg){:class="img-responsive"}

<span style="font-size:1.6em;">**BABS4 - Data Workshop 4**</span><br/>

<p align="justify">
Welcome to Data Workshop 4!<br/>
This is the second of two RNAseq workshops as part of the BABS4 (66I) "Gene expression and biochemical interactions strand". This material carries on directly from the material in workshop 3. If you need to re-do any of that workshop, or just need to refresh your memory, <a href="https://asmasonomics.github.io/courses/BABS4_Biochem_DataWorkshop3_March2024">please follow this link to the previous material</a>.<br/>
<br/>
</p>


# PAGE IN DEVELOPMENT
Course material will be complete before 13th March 2024.

<div class="tenor-gif-embed" data-postid="14031649369780344580" data-share-method="host" data-aspect-ratio="1.52778" data-width="100%"><a href="https://tenor.com/view/hypno-toad-gif-14031649369780344580">Hypno Toad GIF</a>from <a href="https://tenor.com/search/hypno+toad-gifs">Hypno Toad GIFs</a></div> <script type="text/javascript" async src="https://tenor.com/embed.js"></script>

## Reword the introduction
### Introduction
<p align="justify">
In this workshop you will work on publicly available RNAseq data from wildtype <i>Haemophilus influenzae</i>. Remember, this bacterium is naturally competent. During stress, gene networks which regulate competence should be activated to help with survival in difficult conditions.<br/>
In their <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0217255">2020 paper, Black <i>et al.,</i></a> aimed to measure this competence response by switching happily proliferating <i>Haemophilus influenzae</i> in BHI medium, to a "starvation" medium called MIV. They took RNA at multiple time points and had cultures in triplicate to assess the deviation in response and enable statistical testing. We will focus initially on the control (<i>t</i>=0 minutes in MIV; kw20-MIV0) compared with <i>t</i>=30 minutes (kw20-MIV2).<br/>
</p>
![Hi experimental setup](/assets/images/2024-03-11_66I-DW3_Hi_exp_setup.png){:class="img-responsive"} <br/>
<p align="justify">
Hopefully this all sounds quite familiar! If not, after this morning, make time to watch the videos we've generated to support these two workshops. These are embedded below and on the VLE.<br/><br/>
You can also download the introductory slides <a href="/assets/coursefiles/2024-03_66I/66I-DW4_introductory_slides.pdf" download>here as a PDF</a>.<br/>
</p>
[Introduction to transcriptomics](#introduction-to-transcriptomics)<br/>
[How has the data in this workshop been processed so far?](#data-processing-before-this-workshop)<br/>
[What is a sequencing read?](#what-is-a-sequencing-read)
<br/>

### The workshop


































<br/><br/><br/><br/><br/><br/>
[Return to tutorial introduction.](#introduction)<br/>
<br/><br/>

### Video explainers
#### Introduction to transcriptomics
<p align="justify">
This video was included in the week 4 teaching materials on the VLE to view before the workshop - hopefully this is just here as a reminder. Here we introduce you to some of the concepts of transcriptomics - where the data comes from, what it looks like, and what you'll be doing with it in the data workshops.<br/>
</p>
<iframe src="https://york.cloud.panopto.eu/Panopto/Pages/Embed.aspx?id=ff92dbc6-c7a3-47c1-96ac-b1260104d15f&autoplay=false&offerviewer=true&showtitle=true&showbrand=true&captions=true&interactivity=all" height="405" width="100%" style="border: 1px solid #464646;" allowfullscreen allow="autoplay" aria-label="Panopto Embedded Video Player"></iframe>
<br/>
[Return to tutorial introduction.](#introduction)<br/>
[Begin the workshop.](#the-workshop)<br/><br/>

#### Data processing before this workshop
<p align="justify">
This video was included in the week 5 teaching materials on the VLE to support the workshop material. We introduce you to the data you will analyse in these two data workshops (3 and 4), and tell you what has happened to go from raw sequencing data (files for each sample total c.40 million lines) to the counts matrix you will work on (1745 rows and 6 columns, in the first instance).<br/>
</p>
<iframe src="https://york.cloud.panopto.eu/Panopto/Pages/Embed.aspx?id=117ba13f-1f0c-4c2a-8f48-b12a00929d38&autoplay=false&offerviewer=true&showtitle=true&showbrand=true&captions=false&interactivity=all" height="405" width="100%" style="border: 1px solid #464646;" allowfullscreen allow="autoplay" aria-label="Panopto Embedded Video Player"></iframe>
<br/>
[Return to tutorial introduction.](#introduction)<br/>
[Begin the workshop.](#the-workshop)<br/><br/>

#### What is a sequencing read
<p align="justify">
This video was included in the week 6 teaching materials on the VLE to support the workshop material. We had some confusion as to what a sequencing read actually is - so hopefully this explainer helps.<br/>
</p>
<iframe src="https://york.cloud.panopto.eu/Panopto/Pages/Embed.aspx?id=b2f4da0d-6a2c-4e94-a633-b13300fce3cc&autoplay=false&offerviewer=true&showtitle=true&showbrand=true&captions=true&interactivity=all" height="405" width="100%" style="border: 1px solid #464646;" allowfullscreen allow="autoplay" aria-label="Panopto Embedded Video Player"></iframe>
<br/>
[Return to tutorial introduction.](#introduction)<br/>
[Begin the workshop.](#the-workshop)<br/><br/>
